src/test/scala/com/rbmhtechnology/calliope/serializer/kafka/PayloadFormatSerializerSpec.scala

```diff
@@ -67,5 +76,71 @@ class PayloadFormatSerializerSpec extends TestKit(ActorSystem("test"))
         }
       }
     }
+    "a payload not backed by a protcol buffer definition" must {
+      "throw an InvalidProtocolBufferException exception" in {
+        val deserializer = PayloadFormatDeserializer.apply
+
+        intercept[InvalidProtocolBufferException] {
+          deserializer.deserialize("topic", randomBytes)
+        }
+      }
+    }
+    "a PayloadFormat payload not being backed by an akka deserializer" must {
+      "throw an PayloadNotDeserializableException exception" in {
+        val deserializer = PayloadFormatDeserializer.apply
+
+        intercept[PayloadNotDeserializableException] {
+          deserializer.deserialize("topic", payloadFormatWithUnknownManifest.toByteArray)
+        }
+      }
+      "return a Failure with attempt" in {
+        val deserializer = PayloadFormatDeserializer.attempt(identity)
+
+        val t = deserializer.deserialize("topic", payloadFormatWithUnknownManifest.toByteArray)
```

Comment:
can be simplified by using a custom matcher

create a separate file for this
```
trait CustomMatchers {
    def failWith[A <: Throwable](implicit classTag: ClassTag[A]) =
      new TryFailureTypeMatcher[A]()

    case class TryFailureTypeMatcher[A <: Throwable](implicit classTag: ClassTag[A]) extends Matcher[Try[AnyRef]] {
      override def apply(left: Try[AnyRef]): MatchResult = left match {
        case Failure(err) =>
          MatchResult(classTag.runtimeClass.equals(err.getClass),
            s"Failure class [${err.getClass.getName}] did not match expected class [${classTag.runtimeClass.getName}]",
            s"Failure class [${err.getClass.getName}] did match expected class [${classTag.runtimeClass.getName}]"
          )
        case Success(res) =>
          MatchResult(matches = false, s"Unexpected success: $res", s"Unexpected success: $res")
      }
    }
  }
  object CustomMatchers extends CustomMatchers
```

you can then match with
`deserializer.deserialize("topic", payloadFormatWithUnknownManifest.toByteArray) must failWith[PayloadNotDeserializableException]`